lib/pages/resetpassword/reset_password_page.dart

```diff
@@ -0,0 +1,144 @@
+                            .textTheme
+                            .bodyText1
+                            ?.copyWith(color: ColorName.whiteAlpha70),
+                        textAlign: TextAlign.center,
+                      ),
+                    ],
+                  ),
+                ),
+                Form(
+                  key: _formKey,
+                  autovalidateMode: AutovalidateMode.onUserInteraction,
+                  child: _buildForm(),
+                ),
+                Expanded(child: SizedBox.shrink()),
+              ],
+            ),
+          ),
+          SafeArea(
+            child: GestureDetector(
+              onTap: () => context.navigateBack(),
+              child: SizedBox(
+                width: 56,
+                height: 56,
+                child: Assets.icons.icBack.svg(
+                  fit: BoxFit.none,
+                ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildForm() {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        TextFormField(
+          controller: _emailController,
+          decoration: CustomInputDecoration(
+            context: context,
+            hint: context.localization.loginEmail,
+          ),
+          keyboardType: TextInputType.emailAddress,
+          style: Theme.of(context).textTheme.bodyText1,
+          textInputAction: TextInputAction.next,
+          validator: _emailValidator,
+        ),
+        const SizedBox(height: Dimens.defaultMarginPadding),
+        SizedBox(
+          width: double.infinity,
+          child: TextButton(
+            style: ButtonStyle(
+              backgroundColor: MaterialStateProperty.all(Colors.white),
+              foregroundColor: MaterialStateProperty.all(Colors.black),
+              overlayColor: MaterialStateProperty.all(Colors.black12),
+              padding: MaterialStateProperty.all(
+                const EdgeInsets.symmetric(
+                  vertical: Dimens.inputVerticalPadding,
+                ),
+              ),
+              shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+                RoundedRectangleBorder(
+                    borderRadius:
+                        BorderRadius.circular(Dimens.inputBorderRadius)),
+              ),
+              textStyle: MaterialStateProperty.all(
+                Theme.of(context).textTheme.button,
+              ),
+            ),
+            child: Text(context.localization.forgotResetText),
+            onPressed: () => _attemptResetPassword(),
+          ),
+        ),
+      ],
+    );
+  }
+
+  String? _emailValidator(String? value) {
+    if (value == null || !EmailValidator.validate(value)) {
+      return context.localization.validationErrorEmailInvalid;
+    } else {
+      return null;
+    }
+  }
+
+  Future<void> _attemptResetPassword() async {
+    if (_formKey.currentState!.validate()) {
```

Comment:
Just curious, what is the purpose of this line? 🙏 
```
_formKey.currentState!.validate()
```

Reply:
@markgravity I think it's a very basic feature of the Form to validate there is no error before executing the action with values from input https://docs.flutter.dev/cookbook/forms/validation#3-create-a-button-to-validate-and-submit-the-form. 🤔